libs/ojsonnet/Core_jsonnet.ml

```diff
@@ -57,26 +57,28 @@ type ident = string wrap [@@deriving show]
  * no Import (expanded during desugaring).
  * TODO? add Import that resolves lazily during Eval?
  *)
-type expr =
+type expr_with_trace = expr * Tok.t list
```

Comment:
It's a bit heavy because this require to change lots of code, and pattern matching an expression is a bit tedious now
with this pair.
If the problem you had was to find a way to adjust the Tok.t embedded inside AST_jsonnet or Core_jsonnet as you desugar, you could instead use a visitor; We can actually automatically derive those visitors (see what we do in AST_generic.ml) so then you just need to visit the Core_jsonnet AST and adjust all the tokens in one pass.